src/openrct2/ride/Vehicle.cpp

```diff
@@ -7135,21 +7135,21 @@ static const uint8_t TrackTypeToSpinFunction[256] = {
  *
  *  rct2: 0x006D661F
  */
-static void vehicle_update_spinning_car(Vehicle* vehicle)
+void Vehicle::UpdateSpinningCar()
 {
-    if (vehicle->update_flags & VEHICLE_UPDATE_FLAG_ROTATION_OFF_WILD_MOUSE)
+    if (update_flags & VEHICLE_UPDATE_FLAG_ROTATION_OFF_WILD_MOUSE)
     {
-        vehicle->spin_speed = 0;
+        spin_speed = 0;
         return;
     }
 
-    auto vehicleEntry = vehicle->Entry();
+    auto vehicleEntry = Entry();
     if (vehicleEntry == nullptr)
     {
         return;
     }
     int32_t spinningInertia = vehicleEntry->spinning_inertia;
-    int32_t trackType = vehicle->track_type >> 2;
+    int32_t trackType = track_type >> 2;
     int32_t dword_F64E08 = _vehicleVelocityF64E08;
     int32_t spinSpeed;
```

Comment:
i know this is how it currently is but having two variables one spinSpeed and the other spin_speed is just a tad confusing. This one should have maybe been `newSpinSpeed`

Reply:
This variable is actually used for two different uses and it should have also been split into two variables.